heat/core/dndarray.py

```diff
@@ -324,11 +324,25 @@ def get_halo(self, halo_size):
             )
 
         if self.comm.is_distributed() and self.split is not None:
-            min_chunksize = self.shape[self.split] // self.comm.size
-            if halo_size > min_chunksize:
+            # gather lshapes
+            lshape_map = self.create_lshape_map()
+            rank = self.comm.rank
+            size = self.comm.size
+            next_rank = rank + 1
+            prev_rank = rank - 1
+            last_rank = size - 1
+
+            # if local shape is zero and its the last process or the next one is also zero we can exit here
```

Comment:
You want to be checking if a DNDarray is balanced here. Otherwise you could hypothetically have 2 empty processes and then some data on one of the next ones. 
On the other hand, if a DNDarray is balanced and lshape[split] on rank is 0, you don't need to check the next rank.

Reply:
What is the desired behavior if there are empty ranks in the middle? Preventing the halo all together, or skipping the empty ranks and using the next/prev non-empty rank?

Reply:
Hi @lucaspataro,  that's an interesting question. I think  so far we're assuming that operations are performed on balanced DNDarrays, in that case the empty ranks will always be at the end. I would say that for now an empty rank "in the middle" is an edge case we don't really need to worry about. What do you think, @krajsek ?

Reply:
Hi @ClaudiaComito, i added a RuntimeError if tensor is not balanced and remove the check for the next rank like you suggested.